crates/polars-plan/src/plans/aexpr/scalar.rs

```diff
@@ -8,7 +8,7 @@ pub fn is_scalar_ae(node: Node, expr_arena: &Arena<AExpr>) -> bool {
         AExpr::Literal(lv) => lv.is_scalar(),
         AExpr::Function { options, input, .. }
         | AExpr::AnonymousFunction { options, input, .. } => {
-            if options.is_elementwise() {
+            if options.is_elementwise() || !options.flags.contains(FunctionFlags::CHANGES_LENGTH) {
```

Comment:
Ah, yeah.. I feel we need a better flag for this. Now it seems duplicate and tricky. Anyhow. Good fix for now. :+1: